### PR TITLE
feature: temporal interceptors support

### PR DESCRIPTION
**Pull request by cv65kr**
# Reason for This PR

https://github.com/roadrunner-server/roadrunner/issues/1313

## Description of Changes

Telemetry for Temporal
image

PART1: https://github.com/temporalio/roadrunner-temporal/pull/322

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the MIT license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.

### CODE REVIEW

File: plugin.go [hunk 1]
```diff
@@ -46,12 +47,13 @@ type Configurer interface {
...
-	propagators propagation.TextMapPropagator
-	mdw         mdw
-	intcpt      intcpt
+	cfg            *Config
+	log            *zap.Logger
+	tracer         *sdktrace.TracerProvider
+	propagators    propagation.TextMapPropagator
+	mdw            mdw
+	intcpt         intcpt
+	temporalIntcpt temporalIntcpt
```
Comment (rustatian) [inline plugin.go:56]: I guess it's time to realize... we don't pay for the letters 😄 
`intcpt` -> `grpcInterceptor`
`temporalIntcpt` -> `temporalInterceptor`

File: temporal_interceptor.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"github.com/roadrunner-server/sdk/v4/utils"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/trace"
+	"go.temporal.io/sdk/contrib/opentelemetry"
+	"go.temporal.io/sdk/interceptor"
+)
+
+// type alias for the interceptorscd
```
Comment (rustatian) [inline temporal_interceptor.go:11]: What is `interceptorscd` ?

File: plugin.go [hunk 3]
```diff
@@ -131,6 +133,7 @@ func (p *Plugin) Init(cfg Configurer, log Logger) error { //nolint:gocyclo
 	p.propagators = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}, jprop.Jaeger{})
 	p.mdw = httpWrapper(p.propagators, p.tracer, p.cfg.ServiceName)
 	p.intcpt = grpcWrapper(p.propagators, p.tracer)
```
Comment (rustatian) [inline plugin.go:135]: `p.intcpt` -> `p.grpcInterceptor`

**Review by rustatian — CHANGES_REQUESTED** [review]

**Comment by rustatian** [comment]
Hey @cv65kr 👋🏻 

It would be nice to have tests for the new otel functionality (`grpc` and `temporal`). You may use a `stderr` collector and redirect the `stderr` in the buffer like I did here [link](https://github.com/roadrunner-server/rr-e2e-tests/blob/master/plugins/http/http_otlp_test.go#L28). And then check for some otel-specific fields in the output.

File: plugin.go [hunk 4]
```diff
@@ -131,6 +133,7 @@ func (p *Plugin) Init(cfg Configurer, log Logger) error { //nolint:gocyclo
 	p.propagators = propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}, jprop.Jaeger{})
 	p.mdw = httpWrapper(p.propagators, p.tracer, p.cfg.ServiceName)
```
Comment (rustatian) [inline plugin.go:134]: And thus why not to change also my naming 😃 

`p.mdw` -> `p.httpMiddleware`

File: temporal_interceptor.go (hunk 2 above)
Comment (cv65kr) [reply temporal_interceptor.go:11]: Looks like tried to change catalog in terminal and by mistake added it to docblock 😮‍💨

**Comment by cv65kr** [comment]
Fair point about naming - done.

Yep, will write some tests.

**Comment by cv65kr** [comment]
@rustatian updated and tested as well

**Review by rustatian — APPROVED** [review]
LGTM

**Comment by rustatian** [comment]
Nice, thank you @cv65kr 👍🏻

**Pull request merged**